cmd/server-main.go

```diff
@@ -562,26 +520,57 @@ func serverMain(ctx *cli.Context) {
 		logger.LogIf(GlobalContext, err)
 	}
 
-	// Populate existing buckets to the etcd backend
-	if globalDNSConfig != nil {
-		// Background this operation.
-		go initFederatorBackend(buckets, newObject)
-	}
+	go func() {
+		// List buckets to heal, and be re-used for loading configs.
+		buckets, err := newObject.ListBuckets(GlobalContext)
+		if err != nil {
+			logger.LogIf(GlobalContext, fmt.Errorf("Unable to list buckets to heal: %w", err))
+		}
 
-	// Initialize bucket metadata sub-system.
-	globalBucketMetadataSys.Init(GlobalContext, buckets, newObject)
+		if globalIsErasure && len(buckets) > 0 {
+			if len(buckets) == 1 {
+				logger.Info(fmt.Sprintf("Verifying if %d bucket is consistent across drives...", len(buckets)))
+			} else {
+				logger.Info(fmt.Sprintf("Verifying if %d buckets are consistent across drives...", len(buckets)))
+			}
 
-	// Initialize bucket notification sub-system.
-	globalNotificationSys.Init(GlobalContext, buckets, newObject)
+			// Limit to no more than 50 concurrent buckets.
+			g := errgroup.WithNErrs(len(buckets)).WithConcurrency(50)
+			for index := range buckets {
+				index := index
+				g.Go(func() error {
+					_, herr := newObject.HealBucket(GlobalContext, buckets[index].Name, madmin.HealOpts{Recreate: true})
+					return herr
+				}, index)
+			}
+			for _, err := range g.Wait() {
+				if err != nil {
+					logger.LogIf(GlobalContext, fmt.Errorf("Unable to list buckets to heal: %w", err))
+				}
+			}
+		}
 
-	// Initialize site replication manager.
-	globalSiteReplicationSys.Init(GlobalContext, newObject)
+		// Populate existing buckets to the etcd backend
+		if globalDNSConfig != nil {
+			// Background this operation.
+			go initFederatorBackend(buckets, newObject)
+		}
 
-	// Initialize quota manager.
-	globalBucketQuotaSys.Init(newObject)
+		// Initialize bucket metadata sub-system.
+		go globalBucketMetadataSys.Init(GlobalContext, buckets, newObject)
```

Comment:
if globalBucketMetadataSys is not loaded before server starts accepting traffic, there is potential for replication to be skipped on such uploads because replication config is missing - these will not catch up unless existing object replication is enabled. Similarly server side bucket encryption and immediate tiering

Reply:
yeah, we don't really care for such things @poornas